Sources/SwinjectStoryboard.swift

```diff
@@ -158,3 +158,100 @@ public class SwinjectStoryboard: _SwinjectStoryboardBase, SwinjectStoryboardProt
 }
 
 #endif
+
+
+extension SwinjectStoryboard {
+    
+    #if os(iOS) || os(tvOS)
+    
+    private func injectDependency<Arg>(to viewController: UIViewController, arg: Arg) {
+        guard !viewController.wasInjected else { return }
+        defer { viewController.wasInjected = true }
+        
+        let registrationName = viewController.swinjectRegistrationName
+    
+        if let container = container.value as? _Resolver {
+            let option = SwinjectStoryboardOption(controllerType: type(of: viewController))
+            typealias FactoryType = (Resolver, Container.Controller, Arg) -> Container.Controller
+            let _ = container._resolve(name: registrationName, option: option) { (factory: FactoryType) in factory(self.container.value, viewController, arg) }
+        } else {
+            fatalError("A type conforming Resolver protocol must conform _Resolver protocol too.")
+        }
+        
+        for child in viewController.childViewControllers {
+            injectDependency(to: child)
+        }
+    }
```

Comment:
This is a lot of repeated code between methods with different arguments. Can we reuse this code somehow? Alternatively we can generate this similarly to how it is done in Swinject (see. [Container.Arguments.erb](https://github.com/Swinject/Swinject/blob/master/Sources/Container.Arguments.erb))